### DISCUSSION TITLE
FerretDB v2 and Data API

### DISCUSSION THREAD

**Discussion by AlekSi** [FerretDB/FerretDB#4578] [Announcements]
Just a short announcement that following [the deprecation notice by MongoDB](https://www.mongodb.com/products/updates/product-support-deprecation), we started working on [Data API](https://www.mongodb.com/docs/atlas/app-services/data-api/) replacement. The first iteration of that HTTP API should be out in a few weeks, along with the first public version of FerretDB v2. This next major version, built around the PostgreSQL extension, will bring significant compatibility and performance improvements.

**Comment by AlekSi** [comment]
FerretDB v2 with Data API (and [many other improvements](https://blog.ferretdb.io/ferretdb-releases-v2-faster-more-compatible-mongodb-alternative/)) is out!

**Reply by subhadip1203** [reply]
It may be a very stupid question , but in FerretDB v2 with MongoDB  the known differences are only these two ?

1. FerretDB uses the same protocol error names and codes, but the exact error messages may be different in some cases.
2. Collection name must be valid UTF-8.

If that is the case , then it is awsome.

**Reply by AlekSi** [reply]
Those are two differences we don't plan to fix. There might be other differences, but we consider them bugs or missing features and intend to fix/implement them. v2 is _much_ more compatible with MongoDB on features than v1 is.

**Comment by AlekSi** [comment]
https://blog.ferretdb.io/best-alternative-to-mongodb-data-api-and-http-endpoints/
